src/lib/util/headTagToEscapedStrings.ts

```diff
@@ -0,0 +1,33 @@
+import { escapeHtmlString } from '$lib/util/escapeHtmlString';
+import type { HeadTags } from '$lib/components/Head/Head.svelte';
+
+export interface GenericHeadTag {
+	tag: string;
+	name?: string | null;
+	attributes?: Record<string, string | null | undefined> | null;
+	content?: string | null;
+}
+
+export type SupportedHeadTags = HeadTags | GenericHeadTag[];
+
+export const serializeHeadTag = (metaTag: SupportedHeadTags[number]) => {
+	const { tag, attributes, content } = metaTag;
+
+	const serializedAttributes: string[] =
+		attributes && typeof attributes === 'object'
+			? Object.entries(attributes).flatMap(([key, value]) =>
+					value ? `${escapeHtmlString(key)}="${escapeHtmlString(value)}"` : []
+			  )
+			: [];
```

Comment:
I think this is more modern & easier to read than:

```js
const serializedAttrs = [];

if (attrs) {
  for (const attrName in attrs) {
    if (Object.prototype.hasOwnProperty.call(attrs, attrName)) {
      const value = attrs[attrName];
      if (value) {
        serializedAttrs.push(
          `${escapeString(attrName)}="${escapeString(value)}"`,
        );
      }
    }
  }
}
```

Is that OK? Object.entries should already only enumerate only an object's own props, not anything it inherits: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#description

It should supercede both `Object.hasOwnProperty.call()` and `Object.hasOwn()` (es2022)